test/framework/clusterctl/client.go

```diff
@@ -201,6 +202,7 @@ type ConfigClusterInput struct {
 	ControlPlaneMachineCount *int64
 	WorkerMachineCount       *int64
 	Flavor                   string
+	VariableOverwrites       map[string]string
```

Comment:
Let's rename this to `ClusterctlVariables`. This would make it consistent with the change I made through the clusterctl upgrade test (see `UpgradeInput` above) and it makes it very clear that those are not ClusterClass/Cluster variables